File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res=0, buy_val=INT_MAX;
        for(int i=0;i<prices.size();i++){
            buy_val=min(buy_val,prices[i]);
            res=max(res,prices[i]-buy_val);
        }
        return res;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy=-1 
        res = 0 
        for i in range(len(prices)):
            if buy==-1 or prices[i]<prices[buy]: 
                buy = i 
            else: 
                res=max(res,prices[i]-prices[buy])
        return res